packages/components/datepicker/src/components/osds-datepicker/osds-datepicker.tsx

```diff
@@ -0,0 +1,187 @@
+  styleUrl: 'osds-datepicker.scss',
+  shadow: true
+})
+export class OsdsDatepicker implements OdsDatepickerAttribute, OdsDatepickerEvent {
+  controller: OdsDatepickerController = new OdsDatepickerController(this);
+
+  @Element() el!: HTMLElement;
+
+  @State() hasFocus = false;
+  @State() datepickerInstance: Datepicker | undefined = undefined;
+
+  /** Attributes */
+
+  /** @see OdsDatepickerAttribute.clearable */
+  @Prop({ reflect: true }) clearable?: boolean = DEFAULT_ATTRIBUTE.clearable;
+
+  /** @see OdsDatepickerAttribute.disabled */
+  @Prop({ reflect: true }) disabled?: boolean = DEFAULT_ATTRIBUTE.disabled;
+
+  /** @see OdsDatepickerAttribute.error */
+  @Prop({ reflect: true }) error?: string = DEFAULT_ATTRIBUTE.error;
+
+  /** @see OdsDatepickerAttribute.format */
+  @Prop({ reflect: true }) format?: string = DEFAULT_ATTRIBUTE.format;
+
+  /** @see OdsDatepickerAttribute.placeholder */
+  @Prop({ reflect: true }) placeholder?: string = DEFAULT_ATTRIBUTE.placeholder;
+
+  /** @see OdsDatepickerAttribute.value */
+  @Prop({ reflect: true, mutable: true }) value?: Date | undefined | null = DEFAULT_ATTRIBUTE.value;
+
+  /** Events */
+
+  /** @see OdsDatepickerEvents.odsValueChange */
+  @Event() odsValueChange!: EventEmitter<OdsDatepickerValueChangeEventDetail>;
+
+  /** @see OdsDatepickerEvents.odsDatepickerBlur */
+  @Event() odsDatepickerBlur!: EventEmitter<void>;
+
+  /** @see OdsDatepickerEvents.odsDatepickerFocus */
+  @Event() odsDatepickerFocus!: EventEmitter<void>;
+
+  /** Watchers */
+
+  @Watch('value')
+  onValueChange(value: Date, oldValue?: Date) {
+    this.controller.onValueChange(value, oldValue);
+  }
+
+  @Listen('odsValueChange')
+  handleInputValueChange(event: CustomEvent) {
+    if (this.format && event.detail.value.length === this.format.length) {
+      this.onChange(new Date(Datepicker.parseDate(event.detail.value, this.format)))
+    } else if (event.detail.value.length === 0) {
+      this.onChange(null)
+    }
+  }
+
+  /**
+   * @see OdsDatepickerBehavior.emitBlur
+   */
+  emitBlur(): void {
+    this.odsDatepickerBlur.emit();
+  }
+
+  /**
+   * @see OdsDatepickerBehavior.emitFocus
+   */
+  emitFocus(): void {
+    this.odsDatepickerFocus.emit();
+  }
+
+  /**
+   * @see OdsDatepickerBehavior.onBlur
+   */
+  onBlur(): void {
+    this.controller.onBlur();
+  }
+
+  /**
+   * @see OdsDatepickerBehavior.onChange
+   */
+  onChange(newValue: Date | undefined | null): void {
+    this.controller.onChange(newValue, this.value);
+  }
+
+  /**
+   * @see OdsDatepickerBehavior.onFocus
+   */
+  onFocus(): void {
+    this.controller.onFocus();
+  }
+
+  /**
+   * @see OdsInputBehavior.beforeInit
+   */
+  beforeInit() {
+    this.controller.beforeInit();
+  }
+
+  componentWillLoad() {
+    this.beforeInit();
+  }
+
+  componentDidLoad() {
+    if(!this.el.shadowRoot) {
+      return;
+    }
+
+    const hiddenInput = this.el.shadowRoot.querySelector('.osds-datepicker__hidden-input') as HTMLInputElement;
```

Comment:
use an id instead of a class as selector as class needs to be able to change without impacting the component